libs/features/mas/web-components/src/cdt/countdown-timer.js

```diff
@@ -0,0 +1,119 @@
+import { LitElement, html, css } from 'lit';
+import { styles } from './countdown-timer.css.js';
+
+export default class CountdownTimer extends LitElement {
+  static properties = {
+    label: { type: String },
+    daysHoursMins: { type: String },
+    timeRanges: { type: String },
+    daysLeft: { type: String },
+    hoursLeft: { type: String },
+    minutesLeft: { type: String },
+    isVisible: {type: Boolean},
+  };
+
+  #timeWords = [];
+  #timeRangesEpoch = [];
+
+  static styles = [styles];
+
+  constructor(endDateTime) {
+    super();
+    this.label = '';
+    this.daysHoursMins = '';
+    this.timeRanges = '';
+    this.daysLeft = '';
+    this.hoursLeft = '';
+    this.minutesLeft = '';
+    this.isVisible = false;
+  }
+
+  firstUpdated() {
+    this.#timeWords = this.daysHoursMins.split(' ');
+    this.#timeRangesEpoch = this.timeRanges.split(',');
+    this.countdownStart();
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    clearInterval(this.countdownInterval);
+  }
+
+  countdownCompleted() {
+    this.isVisible = false;
+    clearInterval(this.countdownInterval);
+  }
+
+  countdownStart() {
+    this.countdownUpdate();
+    this.countdownInterval = setInterval(() => {
+      this.countdownUpdate();
+    }, 60000);
+  }
+
+  countdownUpdate() {
+
+    let currentTime = Date.now();    
+    //find the time range where the current time is between the start and end time
+    for (let i = 0; i < this.#timeRangesEpoch.length; i += 2) {
+      const startTime = this.#timeRangesEpoch[i];
+      const endTime = this.#timeRangesEpoch[i + 1];
+      
+      if (currentTime >= startTime && currentTime <= endTime) {
+        this.isVisible = true;
+        let diffTime = endTime - currentTime;
+        this.daysLeft = Math.floor(diffTime / (1000 * 60 * 60 * 24));
+        this.hoursLeft = Math.floor((diffTime % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+        this.minutesLeft = Math.floor((diffTime % (1000 * 60 * 60)) / (1000 * 60));
+        this.requestUpdate(); 
```

Comment:
if requestUpdate really needed? when you update reactive properties, it will automatically schedule an update.

Reply:
Yeah, I forgot removing this after making reactive properties